### PR TITLE
Fix XAI algorithm for Detection

### PR DESCRIPTION
**Pull request by GalyaZalesskaya**
### Summary

This PR fixes the issue with saliency maps for ATSS and SSD models with a small number of classes: highlighted background, the problem with small objects detection. 
The comparison between the original approach and the proposed fix:

image
image
image
image

YOLOX model still has a problem with saliency maps for PyTorch and IR models that requires further investigation.

Reported issue with ATSS saliency maps: [CVS-121810](https://jira.devtools.intel.com/browse/CVS-121810)

The fix should be merged to releases/1.5.0 as well, I will make another PR after fixing comments and merging this one

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

**Comment by jaegukhyun** [comment]
Could you check effect of this PR to tiling XAI?

**Comment by GalyaZalesskaya** [comment]
@jaegukhyun Sure, thank you

**Review by goodsong81 — DISMISSED** [review]
Thank you for the fix! Results look far much better.
Quick question: it looks like processing XAI in more larger resolution. How is it related to "small number of class"?

**Comment by GalyaZalesskaya** [comment]
@goodsong81 Hi Songki! The issue with background highlight appears mostly on the small number of classes due to softmax usage for each feature map layer. 
In case of 2 classes, the saliency map for one class becomes an invertion of the other class saliency map as it can be seen on the image.
image

That's why I underline the small number of classes usecase, where moving softmax in the end fixed the issue with background highlight

**Comment by GalyaZalesskaya** [comment]
@jaegukhyun I conducted experiments on ATSS-tiling on the `small_objects` dataset with default tile size of 2377 (that created 4 tiles and one tile as an overall image).
The proposed approach even improved tiling result and its attention to small objects for such a big tile-size: class `rectangle` (Before/After):
image
image

**Review by sovrasov — APPROVED** [review]

**Review by goodsong81 — APPROVED** [review]

**Review by sungmanc — APPROVED** [review]
Thanks for the nice update
